src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -168,6 +276,12 @@ void ConfigManager::parseConfig(const nlohmann::json& j) {
   */
   const auto& jFlattend = j.flatten();
 
+  // All the configuration options together with their paths.
+  std::vector<std::pair<std::string, ConfigOption*>> allConfigOption =
+      configurationOptions("");
+  ad_utility::HashMap<std::string, ConfigOption*> allConfigOptionHashMap(
+      allConfigOption.begin(), allConfigOption.end());
+
```

Comment:
I don't think you need both of tyhem.
Create the HashMap inside an immediately invoked lambda and make it `const`.

Then you do everything on the hashMap (iteration also works).
Then you can call the hashMap `allConfigOptions` (with an `s` for the plural).

Reply:
Clever idea. Done in [3343dcc](https://github.com/ad-freiburg/qlever/pull/1071/commits/3343dcc6c69fe72a584df503d1d86a3f15a3d9d9).